### DISCUSSION TITLE
ROM limit to 24 bits - why?

### DISCUSSION THREAD

**Discussion by GaryMcD** [hneemann/Digital#1196] [Q&A]
Is there a constraint that requires the limit to be 24 bits? I would like to use a rom with a 32 bit address. I have found that I can get around this by opening the .dig file and setting the value to 32 bits, but it will get set back to 24 if I ever use the GUI to change anything else on the rom.

https://github.com/hneemann/Digital/blob/3620e500a45aceaa858511bab8bd38a6761f091f/src/main/java/de/neemann/digital/core/element/Keys.java#L279

**Comment by hneemann** [comment]
What do you want to store in a 4GByte ROM?

**Reply by GaryMcD** [reply]
I admit - a 4Gbyte ROM is a bit excessive. With that said - it seems that all of the memory is limited to 24 address bits, even the RAM. I was working on simulating a RISC-V, 32bit CPU which has 32bits for addressing
